### DISCUSSION TITLE
How to know when placement of floating element changes?

### DISCUSSION THREAD

**Discussion by vitaltechsol** [floating-ui/floating-ui#2490] [Q&A]
Hello,
I'm using `useFloating` with a `FloatingPortal` following the example from `https://codesandbox.io/s/gallant-sea-rcg43b?file=/src/Select.tsx` but I need to know when the floating element changes placement due to scrolling. For example if it pops down but then up after scrolling it pops up. Reason is I want to change the style(shadow) depending on which direction the floating element is placed. Something like 'onPlacementChanged()` does this exist or is it not possible?

Thank you.

**Comment by atomiks** [comment]
> Reason is I want to change the style(shadow) depending on which direction the floating element is placed

`placement` gets returned from the hook

**Comment by vitaltechsol** [comment]
Thank you! that worked (from `useFloating`)
